lib/livebook_web/live/session_live.ex

```diff
@@ -665,11 +666,23 @@ defmodule LivebookWeb.SessionLive do
 
   @impl true
   def handle_params(%{"cell_id" => cell_id}, _url, socket)
-      when socket.assigns.live_action in [:cell_settings, :cell_upload] do
+      when socket.assigns.live_action in [:cell_settings] do
     {:ok, cell, _} = Notebook.fetch_cell_and_section(socket.private.data.notebook, cell_id)
     {:noreply, assign(socket, cell: cell)}
   end
 
+  @impl true
+  def handle_params(%{"section_id" => section_id, "cell_id" => cell_id}, _url, socket)
+      when socket.assigns.live_action in [:cell_upload] do
+    {:noreply, socket |> assign(cell_id: cell_id) |> assign(section_id: section_id)}
```

Comment:
We can aggregate that into `cell_upload_metadata: %{section_id: ..., cell_id: ...}`, since we only send those back from the component :)